framework/helpers/BaseHtml.php

```diff
@@ -1216,26 +1216,7 @@ public static function errorSummary($models, $options = [])
         $encode = ArrayHelper::remove($options, 'encode', true);
         $showAllErrors = ArrayHelper::remove($options, 'showAllErrors', false);
         unset($options['header']);
-
-        $lines = [];
-        if (!is_array($models)) {
-            $models = [$models];
-        }
-        foreach ($models as $model) {
-            /* @var $model Model */
-            foreach ($model->getErrors() as $errors) {
-                foreach ($errors as $error) {
-                    $line = $encode ? Html::encode($error) : $error;
-                    if (!in_array($line, $lines, true)) {
-                        $lines[] = $line;
-                    }
-                    if (!$showAllErrors) {
-                        break;
-                    }
-                }
-            }
-        }
-
+        $lines = Console::collectErrors($models, $encode, $showAllErrors);
```

Comment:
Looks wrong. Html helper should not use Console.

Reply:
Could you please, help me to figure out where can I move it? If I put the same `collectErrors` into each of helpers, codeclimate detects 3 times copypaste. It seems we don't have a class that all `Html`, `Console` and `Json` are shared together. Should I create one?

Reply:
@yiisoft/core-developers, @yiisoft/reviewers any ideas?

Reply:
`collectErrors()` really more related to `Html` than to `Console`
Maybe it's really better to create some... trait, for example?
And basically not use `$encode`, because it relates only to `Html` and use `Html` functional.
And extend it in `Html`, adding `$encode` as third optional parameter with `true` by default.

Maybe it will looks strangely, but will be more right in design terms.
Or say "get off, codeclimate!", and use same code. It will be two times in Console and Json. It's better then to write interface for them, and... it's again story about "right design/simplicity" balance.

Reply:
I'd duplicate it.

Reply:
Done, I duplicated `collectErrors()` for each Helper